runelite-client/src/main/java/net/runelite/client/plugins/pyramidplunder/PyramidPlunderOverlay.java

```diff
@@ -165,4 +190,75 @@ public Dimension render(Graphics2D graphics)
 
 		return null;
 	}
+
+	private void highlightObject(Graphics2D graphics, GameObject object)
+	{
+		ObjectComposition imposter = client.getObjectDefinition(object.getId()).getImpostor();
+
+		if (URN_CLOSED_IDS.contains(imposter.getId())
+			|| GRAND_GOLD_CHEST_CLOSED_ID == imposter.getId()
+			|| SARCOPHAGUS_CLOSED_ID == imposter.getId())
+		{
+			Shape shape = object.getConvexHull();
+
+			if (shape != null)
+			{
+				OverlayUtil.renderPolygon(graphics, shape, config.highlightContainersColor());
+			}
+		}
+	}
+
+	private boolean objectTooFarAway(TileObject object)
+	{
+		LocalPoint playerLocation = client.getLocalPlayer().getLocalLocation();
+
+		return object.getLocalLocation().distanceTo(playerLocation) >= MAX_DISTANCE;
+	}
+
+	private boolean shouldHighlightObjectOnFloor(int objectId, int currentFloor)
+	{
+		if (URN_IDS.contains(objectId))
+		{
+			boolean shouldHighlightPenultimateUrns = config.highlightPenultimateUrns() && URN_IDS.contains(objectId);
+			return
+				currentFloor > this.penultimateFloor && shouldHighlightPenultimateUrns ||
+					currentFloor >= config.highlightUrnsFloor() ||
+					currentFloor == this.penultimateFloor && shouldHighlightPenultimateUrns && stillWithinTimeLimit();
+
+		}
+		else if (GRAND_GOLD_CHEST_ID == objectId)
+		{
+			return currentFloor >= config.highlightChestFloor();
+		}
+		else if (SARCOPHAGUS_ID == objectId)
+		{
+			return currentFloor >= config.highlightSarcophagusFloor();
+		}
+		return false;
+	}
+
+	private Integer getPenultimateFloor()
```

Comment:
This looks like it can be `floor = (level - 1) / 10 -1` which feels less complex.

Reply:
Ah that's a nice clean way of getting the max floor, I'll just use `floor = (level - 1) / 10 -2` and round down

Reply:
I think you should keep the method, so you can handle cases where players are below lvl 21 or have boosted above 100 here.

Reply:
Hm yeah I'll see what I can do with it